Toms-Storage-120/src/shared/java/com/tom/storagemod/gui/AbstractStorageTerminalScreen.java

```diff
@@ -257,19 +257,30 @@ protected void updateSearch() {
 				for (int i = 0;i < getMenu().itemListClient.size();i++) {
 					StoredItemStack is = getMenu().itemListClient.get(i);
 					if (is != null && is.getStack() != null) {
-						String dspName = searchMod ? BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace() : is.getStack().getHoverName().getString();
+						String dspName;
+						if (searchMod) dspName = BuiltInRegistries.ITEM.getKey(is.getStack().getItem()).getNamespace();
+						else if (searchNbt && is.getStack().hasTag()) dspName = String.valueOf(is.getStack().getTag());
+						else if (searchTag) dspName = is.getStack().getTags().toList().toString();
+						else dspName = is.getStack().getHoverName().getString();
 						notDone = true;
 						if (m.matcher(dspName.toLowerCase()).find()) {
 							addStackToClientList(is);
 							notDone = false;
 						}
 						if (notDone) {
-							for (String lp : tooltipCache.get(is)) {
-								if (m.matcher(lp).find()) {
+							if (searchNbt) {
+								if (m.matcher(nbtCache.get(is)).find()) {
 									addStackToClientList(is);
-									notDone = false;
 									break;
 								}
+							} else {
+								List<String> list = searchTag ? tagCache.get(is) : tooltipCache.get(is);
+								for (String lp : list) {
+									if (m.matcher(lp).find()) {
+										addStackToClientList(is);
+										break;
+									}
+								}
```

Comment:
This only runs if the search didn't find a match before, if you are searching by nbt you already check if the nbt maches at https://github.com/tom5454/Toms-Storage/pull/417/files#diff-73a8694f0eeecda5315264597d312a1b58d29f891caa3bade5d19be7d3f5a636L262
Doing it again is useless.